server.js

```diff
@@ -55,208 +56,229 @@ function showHomepage(req, res) {
-                          city = artwork.collecting_institution.slice(artwork.collecting_institution.indexOf(', ') + 2, artwork.collecting_institution.length);
-                        }
+          })
+          //------------------------------------------------------------------------------
+          // Get the results for the search query from the MET's api
+          //------------------------------------------------------------------------------
+          //then, take the array of Artwork objects we created from the Smithsonian superagent call, and send it to get MET results
+          .then(data => {
+            var allArtworks = data;
+            //get the results for the search query from the MET
+            let url = `https://collectionapi.metmuseum.org/public/collection/v1/search?q=${artist}&artistOrCulture`;
+            //call the MET's API which will return a list of object IDs that match the search query.
+            //create a promises array to place all the gets we want based on the returned object IDs, then execute them all at once.
+            var promises = [];
+            superagent.get(url)
+              .then(metData => {
+                var rows = metData.body.objectIDs ? metData.body.objectIDs : [];
+                //for each object ID we get back from the MET query, we now need to create another superagent call to get the details of that object
+                rows.forEach(item => {
+                  //set the url for each item and push the superagent.get call into the promises array
+                  let eachObjectURL = `https://collectionapi.metmuseum.org/public/collection/v1/objects/${item}`;
+                  promises.push(superagent.get(eachObjectURL));
+                });
+                //run the promises array so that we go ahead and call each of the saved superagent.get calls sequentially
+                Promise.all(promises)
+                  .then(data => {
+                    //now we have "data" which is an aggregate of all the results from the superagent.get calls of each artwork
+                    //for each object in data, check whether its artist field matches the search query,
+                    //if it does, then create an object for it and add it to the artworks array.
+                    //if it doesn't, then just ignore that result.
+                    data.forEach(objectData => {
+                      if (objectData.body.artistDisplayName.toLowerCase().indexOf(artist.toLowerCase()) > -1) {
+                        //the user's search query matches the artist's name, so create the object and push it into the allArtworks array.
                         allArtworks.push(new ArtWork(
-                          artwork.collecting_institution, //this is the museum name
-                          artist, //the artist name we got from the previous API call
-                          artwork.title, //the artwork title
```

Comment:
this doesn't look right either..

Reply:
if (artwork.collecting_institution && artwork.collecting_institution.indexOf(', ') > -1) {
                                city = artwork.collecting_institution.slice(artwork.collecting_institution.indexOf(', ') + 2, artwork.collecting_institution.length);
                              }

it looks complete for me, so I'm not sure why part of that is missing.